app/src/main/java/com/example/campusguide/search/CustomSearchActivity.kt

```diff
@@ -7,23 +7,35 @@ import android.view.View
 import android.widget.*
 import androidx.appcompat.app.AppCompatActivity
 import com.example.campusguide.R
+import com.example.campusguide.search.indoor.BuildingIndexSingleton
+import com.example.campusguide.search.indoor.IndoorSearchResultProvider
+import com.example.campusguide.search.outdoor.PlacesApiSearchResultProvider
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
 
-class CustomSearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener, AdapterView.OnItemClickListener {
+class CustomSearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener,
+    AdapterView.OnItemClickListener {
     private lateinit var searchView: SearchView
     private lateinit var listView: ListView
     private lateinit var adapter: SearchResultAdapter
-    private lateinit var searchResultProvider: PlacesApiSearchResultProvider
+    private var searchResultProviders: MutableList<SearchResultProvider> = mutableListOf()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_custom_search)
-        
+
         // Enables the "Back" button to cancel search
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
 
-        searchResultProvider = PlacesApiSearchResultProvider(this)
+        val buildingIndex = BuildingIndexSingleton.getInstance(this.assets)
+        searchResultProviders.add(IndoorSearchResultProvider(buildingIndex, 3))
+        searchResultProviders.add(
+            PlacesApiSearchResultProvider(
+                this,
+                3
```

Comment:
Why only 3? Can't we retrieve all the results that match the query?

Reply:
Good point - I've removed the cap on places API search results